integration_tests/test_suites/daemon-test-suite/auto_run_reexecution_tests/test_auto_run_reexecution.py

```diff
@@ -117,6 +117,37 @@ def test_filter_runs_no_retry_on_asset_or_op_failure(instance_no_retry_on_asset_
         for event in instance.all_logs(run.run_id)
     )
 
+    run = create_run(
+        instance,
+        status=DagsterRunStatus.STARTED,
+        tags={MAX_RETRIES_TAG: "2", RETRY_ON_ASSET_OR_OP_FAILURE_TAG: False},
+    )
+
+    dagster_event = DagsterEvent(
+        event_type_value=DagsterEventType.PIPELINE_FAILURE.value,
+        job_name=run.job_name,
+        message="oops step failure",
+        event_specific_data=JobFailureData(
+            error=None, failure_reason=RunFailureReason.STEP_FAILURE
+        ),
+    )
+    instance.report_dagster_event(dagster_event, run_id=run.run_id, log_level=logging.ERROR)
+
+    # does not retry due to the RETRY_ON_ASSET_OR_OP_FAILURE_TAG tag being false
+
+    assert (
+        len(
+            list(
+                filter_runs_to_should_retry(
```

Comment:
nit: i prefer extracting variables rather than this nesting is more readable in general.